### PR TITLE
feat: support customizing Collapse expandIcon name

### PR DESCRIPTION
**Pull request by jaredleechn**
Need to customize the icon in my case. If the API is ok, I will update the readme soon

#### code
```
<Collapse defaultActiveKey={['1']} onChange={callback} arrowIcon="caret-right">
...
</Collapse>
```

#### appearance
image

### CODE REVIEW

File: components/collapse/Collapse.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ export interface CollapseProps {
   className?: string;
   bordered?: boolean;
   prefixCls?: string;
+  arrowIcon?: string;
```
Comment (picodoth) [inline components/collapse/Collapse.tsx]: 加到 readme 里吧

**Comment by HeskeyBaozi** [comment]
I prefer to pass an React element rather than a literal icon type. e.g.
```jsx
<Collapse
  defaultActiveKey={['1']}
  onChange={callback}
  arrowIcon={<Icon type="caret-right" />}
>
...
</Collapse>
```

**Comment by afc163** [comment]
feature branch please

**Comment by jaredleechn** [comment]
> I prefer to pass a React element rather than a literal icon type.

The Collapse works as following, `renderExpandIcon` hold a className `arrow` for applying style `rotate 90deg` when the panel is open, so if props passed as a reactNode, we need to add another `<Wrapper className="arrow">{switcher}</Wrapper>` for it, which I think is too heavy

```jsx
renderExpandIcon = () => <Icon type={this.props.arrowIcon || 'right'} className={`arrow`} />
render = () => <RcCollapse
  {...this.props}
  className={collapseClassName}
  expandIcon={this.renderExpandIcon}
/>
```

**Comment by jaredleechn** [comment]
In my opinion, it's good to make it scalable for code design. But for Collapse expandIcon, I think that's a little over-design, few people would pass a switcher other than `<Icon />`

Actually my starting point is keeping it simple to use for most cases and less changes to existing code. But if there's a better implement for antd, it's ok for me, feel free to overwrite this branch

**Comment by zombieJ** [comment]
API 统一一下，和 rc-tree 里搞成一样的： https://github.com/react-component/tree/blob/master/src/TreeNode.jsx#L49

@HeskeyBaozi @jaredleechn

**Comment by HeskeyBaozi** [comment]
> In my opinion, it's good to make it scalable for code design. But for Collapse expandIcon, I think that's a little over-design, few people would pass a switcher other than `<Icon />`
> 
> Actually my starting point is keeping it simple to use for most cases and less changes to existing code. But if there's a better implement for antd, it's ok for me, feel free to overwrite this branch

one common case is that the user uses his own custom SVG icon rather than ant design official icons like `right`, `caret-right`.
The function `renderExpandIcon` is only used to render the switcher icon in `rc-collapse`. be free to overwrite it.

**Review by HeskeyBaozi — APPROVED** [review]

**Comment by zombieJ** [comment]
Let's use Component & render function in this to make it flexible. type string is not the best way.

**Comment by yesmeck** [comment]
@jaredleechn

**Comment by jaredleechn** [comment]
> Let's use Component & render function in this to make it flexible. type string is not the best way.

Sorry for the late, but it's hard for me to agree with the above viewpoint, unless we give a similar prop definition like `iconRender` to Button the same

I think scalability is the 1st factor of `react-component`, not `ant-design`

So how about reconsider this MR 😄 @yesmeck @HeskeyBaozi  @zombieJ

**Comment by yesmeck** [comment]
> unless we give a similar prop definition like iconRender to Button the same

WIP in https://github.com/ant-design/ant-design/pull/12888

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/pull/14060

**Comment by hasnainbaloch** [comment]
expandIcon={({ isActive }) => (isActive ? ICON1 : ICON2)}

**Pull request closed without merging**